restapi/resource_api_object.go

```diff
@@ -299,6 +322,17 @@ func resourceRestAPIExists(d *schema.ResourceData, meta interface{}) (exists boo
 	err = obj.readObject()
 	if err == nil {
 		exists = true
+		if enforce, ok := d.GetOk("enforce_data"); ok && enforce.(bool) {
```

Comment:
This makes a lot of sense and is something that I've wanted to add to the provider for quite some time. Unfortunately, this particular implementation falls a bit short because it is only examining first-level child objects to detect differences. I think there needs to be a more "deep" diff that happens

Also, this is expected behavior with Terraform, so we shouldn't require a new parameter.

Reply:
@DRuggeri would you consider accepting just the read_data portion? I have a project I’m working on that needs a POST call when reading. Currently when using the provider and doing a POST on read_method the application will error because no json payload is sent even when the app just expects an empty dictionary.

Would you prefer it gets split off of this pull request to implement this? I can go through the changes and submit a new one with just that.

Reply:
Apologies for the delay, @Xboarder56!

Yes, `read_data` is OK and, in the time since the PR was opened and I've been able to dig into feature reviews again, #252 was opened. Definitely happy to have `read_data`!